packages/client/tests/functional/issues/15644-middleware-arg-types/tests.ts

```diff
@@ -0,0 +1,183 @@
+import testMatrix from './_matrix'
```

Comment:
For these tests, you may be able to use https://jestjs.io/docs/api#testeachtablename-fn-timeout

I think it would look something like: 


```ts
test.each(['create', 'read'])('middlewear with %s', (method) => {
  prisma.$use((params, next) => {
    if (params.action === method) {
      expect(params.args).toStrictEqual({ data: {} })
    }

    return next(params)
  })

  await prisma.resource[method]({ data: {} })
})
```

Reply:
Thanks I'll keep that in mind for next time!